lib/beetle/publisher.rb

```diff
@@ -159,15 +162,16 @@ def bunny?
 
     def new_bunny
       b = Bunny.new(
-        :host           => current_host,
-        :port           => current_port,
-        :logging        => !!@options[:logging],
-        :user           => @client.config.user,
-        :pass           => @client.config.password,
-        :vhost          => @client.config.vhost,
-        :frame_max      => @client.config.frame_max,
-        :channel_max    => @client.config.channel_max,
-        :socket_timeout => @client.config.publishing_timeout,
+        :host               => current_host,
+        :port               => current_port,
+        :logging            => !!@options[:logging],
+        :user               => @client.config.user,
+        :pass               => @client.config.password,
+        :vhost              => @client.config.vhost,
+        :frame_max          => @client.config.frame_max,
+        :channel_max        => @client.config.channel_max,
+        :socket_timeout     => @client.config.publishing_timeout,
+        :connect_timeout    => DEFAULT_BUNNY_CONNECT_TIMEOUT,
```

Comment:
I would just do `:connect_timeout    => DEFAULT_BUNNY_CONNECT_TIMEOUT`. Or introduce a specific connect_timeout config.

Reply:
Our default publishing timeout is 0, so better introduce an additional config option.